azurerm/internal/services/storage/helpers.go

```diff
@@ -4,11 +4,23 @@ import (
 	"context"
 	"fmt"
 	"strings"
+	"sync"
 
 	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
 )
 
+var (
+	accountKeysCache        = map[string]string{}
+	resourceGroupNamesCache = map[string]string{}
```

Comment:
🤔 about this some more; perhaps we want to force-clear this cache when the account's deleted